Source/OrchBase/P_builder.cpp

```diff
@@ -1743,11 +1808,56 @@ unsigned P_builder::WriteThreadVars(string& task_dir, unsigned coreNum,
             // as for the case of inputs, the target device should be replaced by GetHardwareAddress(...)
             unsigned int tgt_idx = ((*device)->par->G.index_a.find(*tgt)->second.to_p)->first;
             P_addr tgt_addr = ((*device)->par->G.index_a.find(*tgt)->second.to_n)->second.data->addr;
-            unsigned tgt_hwaddr = tgt_addr.A_box << (LOG_DEVICES_PER_THREAD + TinselLogThreadsPerCore + TinselLogCoresPerMailbox + TinselLogMailboxesPerBoard + TinselMeshXBits + TinselMeshYBits);
-            tgt_hwaddr |= tgt_addr.A_board << (LOG_DEVICES_PER_THREAD + TinselLogThreadsPerCore + TinselLogCoresPerMailbox + TinselLogMailboxesPerBoard);
-            tgt_hwaddr |= tgt_addr.A_mailbox << (LOG_DEVICES_PER_THREAD + TinselLogThreadsPerCore + TinselLogCoresPerMailbox);
-            tgt_hwaddr |= tgt_addr.A_core << (LOG_DEVICES_PER_THREAD + TinselLogThreadsPerCore);
-            tgt_hwaddr |= tgt_addr.A_thread << (LOG_DEVICES_PER_THREAD);
+            
+            
+            //------------------------------------------------------------------

```

Comment:
What needs to be done to resolve the TODO in these comments :sweet_potato:? Not necessarily now, but I don't know what is blocking the change here.

Reply:
In short, the impending re-write of P_buidler for v4.